aws-datastore/src/main/java/com/amplifyframework/datastore/AWSDataStorePlugin.java

```diff
@@ -551,4 +583,71 @@ public void observe(
             @NonNull Action onObservationCompleted) {
         onObservationFailure.accept(new DataStoreException("Not implemented yet, buster!", "Check back later!"));
     }
+
+    /**
+     * Creates a builder that provides available options to be set when creating
+     * a DataStore plugin.
+     * @return A new instance of the DataStore plugin builder.
+     */
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    /**
+     * Builder object for the DataStore plugin.
+     */
+    public static final class Builder {
+        private DataStoreConfiguration dataStoreConfiguration;
+        private ModelProvider modelProvider;
+        private ModelSchemaRegistry modelSchemaRegistry;
+        private ApiCategory apiCategory;
+
+        private Builder() {}
```

Comment:
```suggestion
        private Builder() {
            modelProvider = ModelProviderLocator.locate();
            modelSchemaRegistry = ModelSchemaRegistry.instance();
            apiCategory = Amplify.API;
        }
```

It would be good to assign default values for them so that `AWSDataStorePlugin.builder().build()` returns what we currently do for `new AWSDataStorePlugin()`.

Reply:
Moving the ModelProviderLocation in here meant also adding a `throws DataStoreException` to the constructor. That wasn't too bad, but for whatever reason, calling the locate method from here was yielding some inconsistent results. 

With that said, those same defaults are applied in the constructor that takes the builder, so the end result of calling `AWSDataStorePlugin.builder().build()` is equivalent to calling `new AWSDataStorePlugin()`